client/collector/src/main/java/org/hawkular/apm/client/collector/DefaultTraceCollector.java

```diff
@@ -856,19 +856,16 @@ public void setFault(String location, String value, String description) {
     @Override
     public void setProperty(String location, String name, String value) {
         if (log.isLoggable(Level.FINEST)) {
-            log.finest("Set business transaction property: location=" + location +
+            log.finest("Set property: location=" + location +
                     " name=" + name + " value=" + value);
         }
 
         try {
             if (fragmentManager.hasFragmentBuilder()) {
                 FragmentBuilder builder = fragmentManager.getFragmentBuilder();
 
-                if (value == null) {
-                    builder.getTrace().getProperties().remove(name);
-                } else {
-                    builder.getTrace().getProperties().add(new Property(name, value));
-                }
+                builder.getCurrentNode().getProperties().add(new Property(name, value));
```

Comment:
The previous code would not work (a bug) as it is trying to remove a string from a set of Property objects. This was a hang over from when getProperties() was a map.

However properties are only added by the collector, and it is possible a property may want to represent a null value, so felt it was better to simplify the code.